pkg/staticnetworkconfig/generator.go

```diff
@@ -167,10 +169,22 @@ func (s *StaticNetworkConfigGenerator) validateAndCalculateMAC(ifaceName string)
 	return strings.Join(splitMac, ":"), nil
 }
 
-func FormatStaticNetworkConfigForDB(staticNetworkConfig []string) string {
+func FormatStaticNetworkConfigForDB(staticNetworkConfig []*models.HostStaticNetworkConfig) string {
 	lines := make([]string, len(staticNetworkConfig))
-	copy(lines, staticNetworkConfig)
+	for i, hostConfig := range staticNetworkConfig {
+		hostLine := hostConfig.NetworkYaml + hostStaticNetworkDelimeter + formatMacInterfaceMap(hostConfig.MacInterfaceMap)
+		lines[i] = hostLine
+	}
 	sort.Strings(lines)
 	// delimeter between hosts config - will be used during nmconnections files generations for ISO ignition
-	return strings.Join(lines, staticNetworkConfigHostDelimeter)
+	return strings.Join(lines, staticNetworkConfigHostsDelimeter)
+}
+
+func formatMacInterfaceMap(macInterfaceMap models.MacInterfaceMap) string {
+	lines := make([]string, len(macInterfaceMap))
+	for i, entry := range macInterfaceMap {
+		lines[i] = fmt.Sprintf("%s=%s", entry.MacAddress, entry.LogicalNicName)
```

Comment:
Should we add validation for the mac-address format in GenerateISO function ?

Reply:
good idea. but we can add it to the swagger, and validate it at the earlier stage